server/datastore/mysql/mysql.go

```diff
@@ -1033,22 +1033,33 @@ func isChildForeignKeyError(err error) bool {
 	return mysqlErr.Number == ER_NO_REFERENCED_ROW_2
 }
 
+type PatternReplacer func(string) string
```

Comment:
I think this is only used internally in the `mysql` package, it should be unexported (`patternReplacer`).